src/Event.php

```diff
@@ -33,8 +34,23 @@ public function acknowledge(Message $message)
         array_map(function ($name) use ($message) {
             $this->emitter->emit($name, $message);
         }, [
-            static::QUEUE_ACKNOWLEDGE,
-            sprintf('%s.%s', static::QUEUE_ACKNOWLEDGE, $message->handler())
+            static::MESSAGE_ACKNOWLEDGE,
+            sprintf('%s.%s', static::MESSAGE_ACKNOWLEDGE, $message->handler())
+        ]);
+    }
+
+    /**
+     * Emits message finished events
+     *
+     * @param Message $message
+     */
+    public function finish(Message $message)
+    {
+        array_map(function ($name) use ($message) {
+           $this->emitter->emit($name, $message) ;
```

Comment:
Will clean this up when this PR is done: https://github.com/thephpleague/event/pull/73